packages/core/src/components/Picker/Picker.tsx

```diff
@@ -1,31 +1,71 @@
 import * as React from "react";
+import {
+  View,
+  StyleSheet,
+  Text,
+  Platform,
+  ViewStyle,
+  StyleProp,
+} from "react-native";
+import { omit, pick, pickBy, identity } from "lodash";
+import { SafeAreaView } from "react-native-safe-area-context";
+import { Picker as NativePicker } from "@react-native-picker/picker";
+
 import { withTheme } from "../../theming";
-//@ts-ignore
-import PickerComponent from "./PickerComponent"; //unable to find file due to using .android/.web/.ios
-import { PickerComponentProps, PickerOption } from "./PickerTypes";
+import Portal from "../Portal/Portal";
+import Button from "../DeprecatedButton";
+import Touchable from "../Touchable";
+import type { Theme } from "../../styles/DefaultTheme";
+import type { IconSlot } from "../../interfaces/Icon";
+import { extractStyles } from "../../utilities";
+
+export interface PickerOption {
+  value: string;
+  label: string;
+}
 
-type Props = PickerComponentProps & {
+export type PickerProps = {
+  error?: any;
   placeholder?: string;
+  disabled?: boolean;
+  style?: StyleProp<ViewStyle> & { height?: number };
   value?: string;
   options: PickerOption[] | string[];
+  onValueChange: (value: string, index: number) => void;
+  defaultValue?: string;
+  assistiveText?: string;
+  label?: string;
+  iconColor?: string;
+  iconSize?: number;
+  leftIconMode?: "inset" | "outset";
+  leftIconName?: string;
+  placeholderTextColor?: string;
+  rightIconName?: string;
+  type?: "solid" | "underline";
+  theme: Theme;
+  Icon: IconSlot["Icon"];
 };
 
-function normalizeOptions(options: Props["options"]): PickerOption[] {
+function normalizeOptions(options: PickerProps["options"]): PickerOption[] {
   if (options.length === 0) {
     return [];
   }
 
-  if (typeof options[0] === "string") {
+  if (typeof options[0] === ("string" || "number")) {
     return (options as string[]).map((option) => ({
-      label: option,
+      label: String(option),
       value: String(option),
     }));
   }
 
-  if (options[0].label && options[0].value) {
-    return options.map((option) => {
+  if (
+    typeof options[0] === "object" &&
```

Comment:
you can use `import { isObject } from 'lodash'` here which could be a tiny bit safer

Reply:
@peterpme Done.